### PR TITLE
fix login error - add jest test

### PR DESCRIPTION
**Pull request by mfka**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -21,3 +21,5 @@ yarn-error.log*
 *.sln
 *.sw*
 
+# jest testing framework
+/coverage
```
Comment (plorencrstit) [inline .gitignore]: maybe it is worth also to add package-json file here?

File: src/security/store.js [hunk 2]
```diff
@@ -60,6 +61,7 @@ const actions = {
       })
         .then(data => {
           localStorage.setItem("token", data.access_token);
+          localStorage.setItem("token_expire", getExpireDate(data.expires_in));
```
Comment (plorencrstit) [inline src/security/store.js]: It is hard to say what `token_expire` is. I think `token_expires_at` would be better and easy to understand it is a date.

File: .gitignore (hunk 1 above)
Comment (szymonpoltorakrst) [reply .gitignore]: `package-lock.json`, `yarn.lock` should be always on repo. The only thing is that you gotta decide which one you use.

File: src/helper/date.helper.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export function getExpireDate(seconds) {
+  const now = new Date().getTime();
+  const date = new Date(now + seconds * 1000);
+  return date.toISOString();
```
Comment (szymonpoltorakrst) [inline src/helper/date.helper.js]: why not moment? `return moment().add(seconds, 'seconds').toISOString();`

File: src/helper/date.helper.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+}
+
+export function isExpired(date) {
+  const expire = new Date(date);
+  return Date.now() >= expire;
```
Comment (szymonpoltorakrst) [inline src/helper/date.helper.js]: same here, `moment().isAfter()` or `moment().isBefore()` depends on your preferences

File: src/security/auth.js [hunk 5]
```diff
@@ -1,6 +1,15 @@
+import { isExpired } from "@/helper/date.helper";
+import store from "@/security/store";
+
 class Auth {
   check() {
```
Comment (szymonpoltorakrst) [inline src/security/auth.js]: I'm not quiet convinced to that name. I saw some approaches like that, but usually it throws error when something is wrong. So you don't use it like `if(check()) {...}` you just call this and wait for it to pass or throw error. this method is basically checking if token is still valid so I suggest something like `isTokenValid`

File: src/security/store.js [hunk 6]
```diff
@@ -73,13 +75,10 @@ const actions = {
...
     localStorage.removeItem("user");
+    localStorage.removeItem("token_expire");
   },
   getUser: (context, token) => {
-    const access_token = token
-      ? token
-      : localStorage.getItem("token")
-        ? localStorage.getItem("token")
-        : false;
+    const access_token = token || localStorage.getItem("token") || false;
```
Comment (szymonpoltorakrst) [inline src/security/store.js]: I think it should be `null` not `false`

File: src/security/store.js [hunk 7]
```diff
@@ -73,13 +75,10 @@ const actions = {
...
     if (access_token) {
       return new Promise((resolve, reject) => {
         UserProvider.provide(access_token)
```
Comment (szymonpoltorakrst) [inline src/security/store.js]: I don't know much about vue approach but it sounds like something is wrong here with Promises. I feel like you unnecessary nesting Promises. We can discuss it later if you want

**Comment by pawlakzn** [comment]
There is some conflict in order.provider.js :(

**Pull request closed without merging**